tests/pr_test.go

```diff
@@ -2,43 +2,84 @@
 package test
 
 import (
+	"log"
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
+	"gopkg.in/yaml.v3"
 )
 
-// Use existing resource group
 const resourceGroup = "geretain-test-resources"
-const defaultExampleTerraformDir = "examples/default"
+const terraformDir = "examples/default"
+
+// Define a struct with fields that match the structure of the YAML data
+const yamlLocation = "../common-dev-assets/common-go-assets/common-permanent-resources.yaml"
+
+type Config struct {
+	SmGuid                  string `yaml:"secretsManagerGuid"`
+	SmRegion                string `yaml:"secretsManagerRegion"`
+	CertificateTemplateName string `yaml:"privateCertTemplateName"`
+}
+
+var smGuid string
+var smRegion string
+var certificateTemplateName string
+
+// TestMain will be run before any parallel tests, used to read data from yaml for use with tests
+func TestMain(m *testing.M) {
+	// Read the YAML file contents
+	data, err := os.ReadFile(yamlLocation)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Create a struct to hold the YAML data
+	var config Config
+	// Unmarshal the YAML data into the struct
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Parse the SM guid and region from data
+	smGuid = config.SmGuid
+	smRegion = config.SmRegion
+	certificateTemplateName = config.CertificateTemplateName
+	os.Exit(m.Run())
+}
 
 func setupOptions(t *testing.T, prefix string, dir string) *testhelper.TestOptions {
+
 	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
 		Testing:       t,
 		TerraformDir:  dir,
 		Prefix:        prefix,
 		ResourceGroup: resourceGroup,
+		TerraformVars: map[string]interface{}{
+			"existing_sm_instance_guid":   smGuid,
+			"existing_sm_instance_region": smRegion,
+			"certificate_template_name":   certificateTemplateName,
+		},
 	})
+
 	return options
 }
 
 func TestRunDefaultExample(t *testing.T) {
 	t.Parallel()
 
-	options := setupOptions(t, "mod-template", defaultExampleTerraformDir)
+	options := setupOptions(t, "sm-private-cert", terraformDir)
 
 	output, err := options.RunTestConsistency()
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
 }
 
 func TestRunUpgradeExample(t *testing.T) {
+	t.Skip()
```

Comment:
I favour skipping the upgrade test by adding the `SKIP UPGRADE TEST` text in the commit message. It does not require creating a subsequent PR to enable the Upgrade test.

Reply:
If we intend to skip only once in this PR. Not applicable to add the above text if we intend to skip upgrade tests for temporalily on this module

Reply:
yea I'm kind of thinking of removing this from the module template too. I have seen a number of repos with the upgrade tests still skipped because a subsequent PR was not raised to enable it. I think we should use the `SKIP UPGRADE TEST` in the commit message for initial PRs to pass

Reply:
Cool - we may want to document that capability in the PR template as well, this is new to me.

Reply:
@ocofaigh @rajatagarwal-ibm - is there documentation on this?

pre-commit complaining.
```
⧗   input: test: Remove skip call. SKIP UPGRADE TEST
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
```

Reply:
I am not sure about the documentation but I give it like this

```
git commit -m "<original message"> -m "SKIP UGPRADE TEST: <why we skipping message">
```

Reply:
Thanks - seems to work anywhere in the message based on the latest run. 👍